kaldb/src/test/java/com/slack/kaldb/logstore/search/KaldbLocalQueryServiceTest.java

```diff
@@ -136,14 +138,18 @@ public void testKalDbSearch() throws IOException {
     assertThat((String) m.source.get("message")).contains("Message100");
 
     // Test histogram buckets
-    assertThat(response.getBucketsList().size()).isEqualTo(2);
-    KaldbSearch.HistogramBucket bucket1 = response.getBuckets(0);
-    assertThat(bucket1.getCount()).isEqualTo(0);
-    assertThat(bucket1.getLow()).isEqualTo(chunk1StartTimeMs);
-    assertThat(bucket1.getHigh()).isEqualTo((chunk1StartTimeMs + chunk1EndTimeMs) / 2.0);
-    KaldbSearch.HistogramBucket bucket2 = response.getBuckets(1);
-    assertThat(bucket2.getCount()).isEqualTo(1);
-    assertThat(bucket2.getHigh()).isEqualTo(chunk1EndTimeMs);
+    InternalAutoDateHistogram dateHistogram =
```

Comment:
all these tests end up with the use-case where there is only 1 chunk. Do we have a test with multiple chunks so that the `reduce` also gets tested? [SearchResultAggregatorImplTest.java](https://github.com/slackhq/kaldb/pull/465/files#diff-6070f57fcffa18fd3e9116c27274b1b84c1f90443868f33337cb0fc74c500417) maybe?

Reply:
In the class you identified we attempt to run an `aggregate` on multiple search results, which results in the internal `reduce`  logic being invoked across multiple search result sets.